packages/@aws-cdk/aws-elasticloadbalancing/lib/load-balancer.ts

```diff
@@ -226,7 +231,7 @@ export class LoadBalancer extends Resource implements IConnectable {
     this.connections = new Connections({ securityGroups: [this.securityGroup] });
 
     // Depending on whether the ELB has public or internal IPs, pick the right backend subnets
-    const subnets: ISubnet[] = props.internetFacing ? props.vpc.publicSubnets : props.vpc.privateSubnets;
+    const subnets: ISubnet[] = loadBalancerSubnets(props);
```

Comment:
Better to make this function return `SelectedSubnets`. Instead of mapping over the `ISubnet` objects, use `selectedSubnets.subnetIds`, and the other members on that structure, to pass to the load balancer below.